### PR TITLE
Added a more consistent logging format

### PR DESCRIPTION
**Pull request by mgleung**
## Description
Added a new logging format to follow the same structure as logs in Felix to provide more information in a more easily understood way

Requires https://github.com/projectcalico/libcalico-go/pull/474

Fixes https://github.com/projectcalico/cni-plugin/issues/236

## Testing
- Ran the unit tests and verified the logs are formatted appropriately

### CODE REVIEW

File: calico_cni_k8s_test.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"syscall"
 	"time"
 
+	"github.com/Sirupsen/logrus"
```
Comment (gunjan5) [inline calico_cni_k8s_test.go]: I think you should be able to get rid of the `"log"` import and use `logrus` everywhere and import it (logrus) as `log` to be consistent with other files

File: calico_cni_k8s_test.go [hunk 2]
```diff
@@ -30,6 +32,8 @@ import (
 func init() {
 	// Create a random seed
 	rand.Seed(time.Now().UTC().UnixNano())
+	logrus.SetFormatter(&logutils.Formatter{})
```
Comment (gunjan5) [inline calico_cni_k8s_test.go]: don't you also need to do this for other `_test.go` files?

File: ipam/calico-ipam.go [hunk 3]
```diff
@@ -30,13 +30,19 @@ import (
...
+	"github.com/projectcalico/libcalico-go/lib/logutils"
 	cnet "github.com/projectcalico/libcalico-go/lib/net"
 )
 
 // VERSION is filled out during the build process (using git describe output)
 var VERSION string
 
 func main() {
+	// Set up logging formatting.
+	log.SetFormatter(&logutils.Formatter{})
```
Comment (gunjan5) [inline ipam/calico-ipam.go]: will these config options also be applicable to the context loggers we have in this file? for example line 91, line 104, etc.? (same for other files)

File: calico_cni_k8s_test.go (hunk 2 above)
Comment (mgleung) [inline calico_cni_k8s_test.go]: I didn't see context loggers being created in the other test files and since they don't output any log output (at least from the tests I ran), I didn't think it was necessary to set up formatting for them.

File: ipam/calico-ipam.go (hunk 3 above)
Comment (mgleung) [inline ipam/calico-ipam.go]: Yup, once its set, it is global

File: glide.lock [hunk 4]
```diff
@@ -126,7 +139,8 @@ imports:
 - name: github.com/projectcalico/go-yaml-wrapper
   version: 598e54215bee41a19677faa4f0c32acd2a87eb56
 - name: github.com/projectcalico/libcalico-go
-  version: 8ddc30eb43327268263d3e0629c62f916d2bd2ee
+  version: 97655b408c5aa0afd43639bc65a7d5b35346ad5a
+  repo: https://github.com/mgleung/libcalico-go
```
Comment (mgleung) [inline glide.lock]: Made a mistake, glide.yaml didn't get committed for some reason

File: glide.lock [hunk 5]
```diff
@@ -1,6 +1,10 @@
-hash: 3da1768d0bbd08052225cfc5391808133a73cc88db3f3366269467af2a7d07df
```
Comment (mgleung) [inline glide.lock]: my glide.yaml file didn't get committed for some reason.  I'll fix that.

File: calico_cni_k8s_test.go (hunk 2 above)
Comment (gunjan5) [inline calico_cni_k8s_test.go]: what about https://github.com/projectcalico/cni-plugin/blob/master/calico_cni_test.go#L5 ?

**Comment by gunjan5** [comment]
also, have you tested this with journalctl?

**Comment by mgleung** [comment]
Tested it in a cluster and viewed logs with journalctl.  Format is as expected

**Comment by mgleung** [comment]
Example log lines:
```
Jul 20 22:28:55 n1 kubelet[1537]: Calico CNI releasing IP address
Jul 20 22:28:55 n1 kubelet[1537]: 2017-07-20 22:28:55.643 [INFO][5201] utils.go 250: Configured environment: [PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin INVOCATION_ID=f647427b58d64c4aa
Jul 20 22:28:55 n1 kubelet[1537]: 2017-07-20 22:28:55.644 [INFO][5201] client.go 202: Loading config from environment
Jul 20 22:28:55 n1 kubelet[1537]: 2017-07-20 22:28:55.645 [INFO][5201] calico-ipam.go 220: Releasing address using workloadID Workload="advanced-policy-demo.nginx-3181653891-mx0qg"
Jul 20 22:28:55 n1 kubelet[1537]: 2017-07-20 22:28:55.645 [INFO][5201] ipam.go 738: Releasing all IPs with handle 'advanced-policy-demo.nginx-3181653891-mx0qg'
```

File: calico_cni_k8s_test.go (hunk 2 above)
Comment (mgleung) [inline calico_cni_k8s_test.go]: Same thing.  No context logger is called so its not necessary to instantiate the formatter.

Comment (gunjan5) [inline calico_cni_k8s_test.go]: I thought you're also switching out `"log"` with logrus in this PR. That's what I meant by that comment :), it's ok if you want to do it in a separate PR if there's a bunch more files with `"log"` import

Comment (mgleung) [inline calico_cni_k8s_test.go]: oh crap, you're right.  My bad didn't read carefully enough.  Since its only a couple of files, I'll fix them up with this PR
